### PR TITLE
WIP: Scaleway init dns

### PR DESCRIPTION
**Pull request by Mia-Cross**
Adding support for Scaleway: Step 1: DNS

### CODE REVIEW

File: vendor/github.com/scaleway/scaleway-sdk-go/scw/version.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+import (
+	"fmt"
+	"runtime"
+)
+
+// TODO: versioning process
+const version = "v1.0.0-beta.7+dev"
+
+var userAgent = fmt.Sprintf("scaleway-sdk-go/%s (%s; %s; %s)", version, runtime.Version(), runtime.GOOS, runtime.GOARCH)
```
Comment (remyleone) [inline vendor/github.com/scaleway/scaleway-sdk-go/scw/version.go:11]: Precise here that we are using kops and that it is an embedded version

File: vendor/github.com/scaleway/scaleway-sdk-go/scw/env.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	terraformAccessKeyEnv    = "SCALEWAY_ACCESS_KEY" // used both as access key and secret key
+	terraformSecretKeyEnv    = "SCALEWAY_TOKEN"
+	terraformOrganizationEnv = "SCALEWAY_ORGANIZATION"
+	terraformRegionEnv       = "SCALEWAY_REGION"
+	cliTLSVerifyEnv          = "SCW_TLSVERIFY"
+	cliOrganizationEnv       = "SCW_ORGANIZATION"
+	cliRegionEnv             = "SCW_REGION"
+	cliSecretKeyEnv          = "SCW_TOKEN"
+
+	// TBD
```
Comment (remyleone) [inline vendor/github.com/scaleway/scaleway-sdk-go/scw/env.go:36]: Can be cleaned up :)

File: vendor/github.com/scaleway/scaleway-sdk-go/scw/env.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	ScwActiveProfileEnv         = "SCW_PROFILE"
+	ScwAPIURLEnv                = "SCW_API_URL"
+	ScwInsecureEnv              = "SCW_INSECURE"
+	ScwDefaultOrganizationIDEnv = "SCW_DEFAULT_ORGANIZATION_ID"
+	ScwDefaultProjectIDEnv      = "SCW_DEFAULT_PROJECT_ID"
+	ScwDefaultRegionEnv         = "SCW_DEFAULT_REGION"
+	ScwDefaultZoneEnv           = "SCW_DEFAULT_ZONE"
+	DebugEnv                    = logger.DebugEnv
+
+	// All deprecated (cli&terraform)
```
Comment (remyleone) [inline vendor/github.com/scaleway/scaleway-sdk-go/scw/env.go:26]: Can be cleaned up too

File: vendor/github.com/scaleway/scaleway-sdk-go/scw/config_legacy.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+	"os"
+	"path/filepath"
+
+	"github.com/scaleway/scaleway-sdk-go/internal/errors"
+	"github.com/scaleway/scaleway-sdk-go/logger"
+	"gopkg.in/yaml.v2"
+)
+
+// configV1 is a Scaleway CLI configuration file
+type configV1 struct {
```
Comment (remyleone) [inline vendor/github.com/scaleway/scaleway-sdk-go/scw/config_legacy.go:15]: Can be cleaned up

File: vendor/github.com/scaleway/scaleway-sdk-go/logger/default_logger.go [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+)
+
+var DefaultLogger = newLogger(os.Stderr, LogLevelWarning)
```
Comment (remyleone) [inline vendor/github.com/scaleway/scaleway-sdk-go/logger/default_logger.go:12]: Would it be a good idea to use klog here?

**Comment by Mia-Cross** [comment]
This pull request is starting to get old and I'm instead going to make several smaller ones so I should close it.

**Pull request closed without merging**